alerting/provider/pagerduty/pagerduty.go

```diff
@@ -13,22 +13,55 @@ const (
 	restAPIURL = "https://events.pagerduty.com/v2/enqueue"
 )
 
+type Integrations struct {
+	IntegrationKey string `yaml:"integration-key"`
+	Group          string `yaml:"group"`
+}
+
 // AlertProvider is the configuration necessary for sending an alert using PagerDuty
 type AlertProvider struct {
 	IntegrationKey string `yaml:"integration-key"`
 
 	// DefaultAlert is the default alert configuration to use for services with an alert of the appropriate type
 	DefaultAlert *alert.Alert `yaml:"default-alert"`
+
+	Integrations []Integrations `yaml:"integrations"`
 }
 
 // IsValid returns whether the provider's configuration is valid
 func (provider *AlertProvider) IsValid() bool {
-	return len(provider.IntegrationKey) == 32
+	groups := make(map[string]int)
+	if provider.Integrations != nil {
+		for _, integration := range provider.Integrations {
+			if _, present := groups[integration.Group]; present || integration.Group == "" || len(integration.IntegrationKey) != 32 {
+				return false
+			}
+			groups[integration.Group] = 1
```

Comment:
```suggestion
	registeredGroups := make(map[string]bool)
	if provider.Integrations != nil {
		for _, integration := range provider.Integrations {
			if isAlreadyRegistered := registeredGroups[integration.Group]; isAlreadyRegistered || integration.Group == "" || len(integration.IntegrationKey) != 32 {
				return false
			}
			registeredGroups[integration.Group] = true
```

For the sake of making this a little easier to read